creator-node/src/utils.js

```diff
@@ -484,20 +486,22 @@ async function createDirForFile (fileStoragePath) {
   await fs.ensureDir(dir)
 }
 
-async function writeStreamToFileSystem (stream, expectedStoragePath, createDir = false) {
+async function writeStreamToFileSystem (inputStream, expectedStoragePath, createDir = false) {
   if (createDir) {
     await createDirForFile(expectedStoragePath)
   }
 
-  const destinationStream = fs.createWriteStream(expectedStoragePath)
-  stream.pipe(destinationStream)
-  return new Promise((resolve, reject) => {
-    destinationStream.on('finish', () => {
-      resolve()
-    })
-    destinationStream.on('error', err => { reject(err) })
-    stream.on('error', err => { destinationStream.end(); reject(err) })
-  })
+  await streamFileToDiskHelper(inputStream, expectedStoragePath)
+}
+
+async function streamFileToDiskHelper (inputStream, expectedStoragePath) {
+  // https://nodejs.org/en/docs/guides/backpressuring-in-streams/
```

Comment:
smallnit can we move all these comments to jsdoc for the function

Reply:
hmm actually, I think it is useful to keep doc references above the line of code that may require additional info. doesn't really matter for me in this scenario bc the fn is small

Reply:
yeah i meant for the below two comment lines, agree with you for this link